libapicore/ApiServer.cpp

```diff
@@ -787,7 +787,7 @@ Json::Value ApiConnection::getMinerStat1()
 {
     auto runningTime = std::chrono::duration_cast<std::chrono::minutes>(
         steady_clock::now() - m_farm.farmLaunched());
-
+    auto connection = m_mgr.getActiveConnection();
```

Comment:
This is bad. getActiveConnection returns a pointer to an object with no guarantee that the object will remain valid.

Reply:
Will try to solve ...

Reply:
... and don't like locks ...